app/jobs/create_solr_collection_job.rb

```diff
@@ -9,18 +9,21 @@ class CreateSolrCollectionJob < ApplicationJob
   def perform(account)
     name = account.tenant.parameterize
 
-    unless collection_exists? name
-      client.get '/solr/admin/collections', params: collection_options.merge(action: 'CREATE',
-                                                                             name: name)
-    end
+    perform_for_cross_search_tenant(account, name) if account.is_shared_search_enabled? && account.tenant_list.present?
+    perform_for_normal_tenant(account, name) unless account.is_shared_search_enabled? && account.tenant_list.present?
```

Comment:
I think an if/else might be better here. I've read this 3 times and I'm pretty sure that if you number these methods like so

1 = perform_for_cross_search_tenant
2 = perform_for_normal_tenant
3 = neither one is done

and make a truth table you get:

tenant_list.present? && is_shared_search_enabled?  => 1
just is_shared_search_enabled?  => 3
just tenant_list.present?  => 3
neither tenant_list.present? or is_shared_search_enabled? => 2

It doesn't seem right to me that would be the case.

```suggestion
    if account.is_shared_search_enabled? && account.tenant_list.present?
      perform_for_cross_search_tenant(account, name)
    else
      perform_for_normal_tenant(account, name)
    end
```

Reply:
Thanks that change has been applied. Thanks